File: app/src/main/java/com/kotlin/loyal/base/BaseClickHandler.java
package com.kotlin.loyal.base;

import android.app.ProgressDialog;
import android.content.Intent;
import android.databinding.ViewDataBinding;
import android.os.Bundle;
import android.support.annotation.StringRes;
import android.view.Window;
import android.view.WindowManager;

import com.mwm.loyal.imp.Contact;
import com.mwm.loyal.utils.StringUtil;
import com.mwm.loyal.utils.ToastUtil;

public abstract class BaseClickHandler<V extends ViewDataBinding> implements Contact {
    protected ProgressDialog progressDialog;
    protected BaseActivity activity;
    protected V binding;

    public BaseClickHandler(BaseActivity baseActivity) {
        this(baseActivity, null);
    }

    public BaseClickHandler(BaseActivity baseActivity, V binding) {
        this.activity = baseActivity;
        this.binding = binding;
        initDialog(baseActivity);
    }

    public final String getString(@StringRes int resId) {
        return activity.getResources().getString(resId);
    }

    public final void showToast(@StringRes int resId) {
        ToastUtil.showToast(activity, resId);
    }

    public final void showToast(CharSequence sequence) {
        ToastUtil.showToast(activity, replaceNull(sequence));
    }

    public final void showToast(String text) {
        ToastUtil.showToast(activity, replaceNull(text));
    }

    public final void showErrorDialog(String text, boolean finish) {
        StringUtil.showErrorDialog(activity, replaceNull(text), finish);
    }

    public final void showDialog(String text, boolean finish) {
        ToastUtil.showDialog(activity, replaceNull(text), finish);
    }

    public final void startActivity(Intent intent) {
        activity.startActivity(intent);
    }

    public final void startActivityForResult(Intent intent, int reqCode) {
        activity.startActivityForResult(intent, reqCode);
    }

    public final void startActivityForResult(Intent intent, int reqCode, Bundle bundle) {
        activity.startActivityForResult(intent, reqCode, bundle);
    }

    public final void finish() {
        activity.finish();
    }

    public final void setResult(int resultCode) {
        activity.setResult(resultCode);
    }

    public final void setResult(int resultCode, Intent intent) {
        activity.setResult(resultCode, intent);
    }

    protected final String replaceNull(Object object) {
        return StringUtil.replaceNull(object);
    }

    private void initDialog(BaseActivity baseActivity) {
        progressDialog = new ProgressDialog(baseActivity);
        progressDialog.setMessage("处理中...");
        Window window = progressDialog.getWindow();
        if (window != null) {
            WindowManager.LayoutParams lp = window.getAttributes();
            lp.alpha = 0.7f;// 透明度
            lp.dimAmount = 0.8f;// 黑暗度
            window.setAttributes(lp);
        }
        progressDialog.setCancelable(false);
        progressDialog.setCanceledOnTouchOutside(false);
    }

    public void showDialog() {
        if (null != progressDialog)
            progressDialog.show();
    }

    public void disMissDialog() {
        if (null != progressDialog && progressDialog.isShowing())
            progressDialog.dismiss();
    }
}

File: app/src/main/java/com/kotlin/loyal/base/BaseClickHandler.kt
package com.kotlin.loyal.base

import android.app.ProgressDialog
import android.content.Intent
import android.databinding.ViewDataBinding
import android.os.Bundle
import android.support.annotation.StringRes
import com.kotlin.loyal.impl.Contact
import com.kotlin.loyal.utils.StringUtil
import com.kotlin.loyal.utils.ToastUtil

abstract class BaseClickHandler<V : ViewDataBinding> @JvmOverloads constructor(protected var activity: BaseActivity<*>, protected var binding: V? = null) : Contact {
    protected var progressDialog: ProgressDialog? = null

    init {
        initDialog(activity)
    }

    fun getString(@StringRes resId: Int): String {
        return activity.resources.getString(resId)
    }

    fun showToast(@StringRes resId: Int) {
        ToastUtil.showToast(activity, resId)
    }

    fun showToast(sequence: CharSequence) {
        ToastUtil.showToast(activity, replaceNull(sequence))
    }

    fun showToast(text: String) {
        ToastUtil.showToast(activity, replaceNull(text))
    }

    fun showErrorDialog(text: String, finish: Boolean) {
        StringUtil.showErrorDialog(activity, replaceNull(text), finish)
    }

    fun showDialog(text: String, finish: Boolean) {
        ToastUtil.showDialog(activity, replaceNull(text), finish)
    }

    fun startActivity(intent: Intent) {
        activity.startActivity(intent)
    }

    fun startActivityForResult(intent: Intent, reqCode: Int) {
        activity.startActivityForResult(intent, reqCode)
    }

    fun startActivityForResult(intent: Intent, reqCode: Int, bundle: Bundle) {
        activity.startActivityForResult(intent, reqCode, bundle)
    }

    fun finish() {
        activity.finish()
    }

    fun setResult(resultCode: Int) {
        activity.setResult(resultCode)
    }

    fun setResult(resultCode: Int, intent: Intent) {
        activity.setResult(resultCode, intent)
    }

    protected fun replaceNull(`object`: Any): String {
        return StringUtil.replaceNull(`object`)
    }

    private fun initDialog(baseActivity: BaseActivity<*>) {
        progressDialog = ProgressDialog(baseActivity)
        progressDialog!!.setMessage("处理中...")
        val window = progressDialog!!.window
        if (window != null) {
            val lp = window.attributes
            lp.alpha = 0.7f// 透明度
            lp.dimAmount = 0.8f// 黑暗度
            window.attributes = lp
        }
        progressDialog!!.setCancelable(false)
        progressDialog!!.setCanceledOnTouchOutside(false)
    }

    fun showDialog() {
        if (null != progressDialog)
            progressDialog!!.show()
    }

    fun disMissDialog() {
        if (null != progressDialog && progressDialog!!.isShowing)
            progressDialog!!.dismiss()
    }
}
